perception/autoware_object_merger/src/object_association_merger_node.cpp

```diff
@@ -55,6 +55,34 @@ bool isUnknownObjectOverlapped(
   return precision > precision_threshold || recall > recall_threshold ||
          generalized_iou > generalized_iou_threshold;
 }
+
+bool isKnownObjectOverlapped(
+  const autoware_perception_msgs::msg::DetectedObject & object1,
+  const autoware_perception_msgs::msg::DetectedObject & object2, const double precision_threshold,
+  const double recall_threshold, const std::map<int, double> & distance_threshold_map,
+  const std::map<int, double> & generalized_iou_threshold_map)
+{
+  int object1_label = object_recognition_utils::getHighestProbLabel(object1.classification);
+  int object2_label = object_recognition_utils::getHighestProbLabel(object2.classification);
+
+  const double generalized_iou_threshold = std::min(
+    generalized_iou_threshold_map.at(object1_label),
+    generalized_iou_threshold_map.at(object2_label));
+  const double distance_threshold =
+    std::min(distance_threshold_map.at(object1_label), distance_threshold_map.at(object2_label));
+
+  const double sq_distance_threshold = std::pow(distance_threshold, 2.0);
+  const double sq_distance = autoware::universe_utils::calcSquaredDistance2d(
+    object1.kinematics.pose_with_covariance.pose, object2.kinematics.pose_with_covariance.pose);
+  if (sq_distance_threshold < sq_distance) return false;
+
+  const auto precision = object_recognition_utils::get2dPrecision(object1, object2);
```

Comment:
Please use a different set of parameters for thresholds.

Reply:
- Added param for known objects.

- Added unknown keywords to existing parameter names.